rllib/agents/qmix/qmix_policy.py

```diff
@@ -226,133 +257,170 @@ def compute_actions(self,
                         info_batch=None,
                         episodes=None,
                         **kwargs):
-        obs_batch, action_mask = self._unpack_observation(obs_batch)
+        obs_batch, action_mask, _ = self._unpack_observation(obs_batch)
+        # We need to ensure we do not use the state to compute actions
 
         # Compute actions
-        with th.no_grad():
-            q_values, hiddens = _mac(
-                self.model, th.from_numpy(obs_batch),
-                [th.from_numpy(np.array(s)) for s in state_batches])
-            avail = th.from_numpy(action_mask).float()
-            masked_q_values = q_values.clone()
-            masked_q_values[avail == 0.0] = -float("inf")
-            # epsilon-greedy action selector
-            random_numbers = th.rand_like(q_values[:, :, 0])
-            pick_random = (random_numbers < self.cur_epsilon).long()
-            random_actions = Categorical(avail).sample().long()
-            actions = (pick_random * random_actions +
-                       (1 - pick_random) * masked_q_values.max(dim=2)[1])
-            actions = actions.numpy()
-            hiddens = [s.numpy() for s in hiddens]
+        with self.lock:
```

Comment:
I guess the lock is needed for APEX mode only, is that right?

Reply:
I was just copying what was done in the TorchPolicy class. Are the locks needed in the read operations?